src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -82,4 +82,20 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+
+    /**
+     * Get a printable string consists of the given printables
+     * @param printables
+     * @return a printable string composed of given printables
+     */
+    default String getPrintableString(Printable... printables) {
+
+        String printableString = " ";
+
+        for (Printable printable : printables) {
+            printableString += printable.getPrintableString() + " ";
```

Comment:
:bulb: *Java tip*: Java 8 has a `StringJoiner` class that can join multiple strings using a delimiter e.g. `aaa` `bbb` `ccc` --> `aaa, bbb, ccc` https://www.mkyong.com/java8/java-8-stringjoiner-example/